README.md

```diff
@@ -100,6 +100,89 @@ On the right is the Central E-SOH API Endpoint. In the middle are all the compon
+
+##### Files-based storage
+​
+Files-based storage was considered, but as query logic has to be implemented separately and there is requirements for structured data and queries, we do not consider it as a feasible option for implementation. There's also another downside with files which is challenge to update minor changes (e.g. incoming QC updates some minutes after the initial observation time) in the middle of the files. When using files, there are typically also some challenges with locking while writing -- in databases there are transactions to handle this// 2c
+ better for multiple calls.
+​
+If files-based datasets, e.g. 2d/3d imaging data or similar, needs to be handled, the support can be added e.g. by integrating S3 object storage into the system and using references to file URLs from the database. Database system will then serve also as a (discovery) metadata backend for files.
+
+
+#### Event sourcing architecture scenario
+
+The specification included a requirement for the Replay functionality, which means that a  user can request missed notification messages for a certain period. Thus we looked into an architecture which needed to have all notifications stored in some way
+
+If we assume that the notification messages also have the actual data (and metadata),
+an [event sourcing](https://www.martinfowler.com/eaaDev/EventSourcing.html) architecture might be appropriate:
+
+- All incoming data is processed and translated into the notification event format, probably some kind of data.
+- All these events are stored in an event store. This is the source of truth.
+- The Replay API is built on top of the event store, and simply returns all notifications in a specific time range.
+- The events are also ingested into the "view" database, which is designed to allow efficient EDR queries.
+- The "view" database can be rebuilt from scratch using the event stream (through the Replay API).
+
+wis2box uses Elasticsearch for storing the events for the Replay API, basically using Elasticsearch
+as a document store. This choice makes sense for wis2box, as Elastischsearch is alreay in the stack to
+support OGC Feature queries. An alternative is to use PostgreSQL as an event store.
+
+An example C4 diagram using Elasticsearch as event store is given below.
+
+![](https://raw.githubusercontent.com/EURODEO/e-soh-c4/main/04-component-diagrams/datastore-options/c4-container-elastic-data-store-event.png)
+
+Based on further architecture discussions it was decided 25th of April 2023 at the Helsinki workshop, that the Replay-functionality will not be implemented.
```

Comment:
I suggest to modify a bit, as several team members were not present, and since we may not fully understand the consequence of this:
```suggestion
Based on further architecture discussions it was suggested 25th of April 2023 at the Helsinki workshop, that the Replay-functionality will not be implemented.
```